### DISCUSSION TITLE
Jira Epic as an issue type in Issue Filter

### DISCUSSION THREAD

**Discussion by MaximumTrainer** [nbrown02/FlowViz-Jira#38] [Q&A] [answered]
In my implementation of flowviz I can see various issue types including story & custom issue types, but I cannot see jira epics & I cannot find where this is configured in the powerbi scripts. 
Can you advise what I need to change to show jira epics in the issue type filter.

**Comment by nbrown02** [comment, marked as answer]
Hi there,

If you go into Power Query and you check the source, you will see for each table it specifies a 'hierarchylevel' - 
![image](https://github.com/nbrown02/FlowViz-Jira/assets/29369962/02ab1734-f712-4d7a-90d6-fd88fdbb487d)

[Jira's documentation](https://support.atlassian.com/jira-cloud-administration/docs/configure-the-issue-type-hierarchy/) explains how "By default, Jira is set up with 3 levels of issue type hierarchy: Epic (level 1), Story (level 0) and Sub-Task (level -1)"

You would need to change the source for every table to account for multiple hierarchy levels.

So, for example, if you wanted to include Epics, in addition to Stories, you would need to change from this:
_AND%20hierarchyLevel%20%3D%200%20_

To:
_AND%20hierarchyLevel%20%21%3D%20-1%20_

You MUST also ensure you change this for any StartAt functions too :)

**Reply by MaximumTrainer** [reply]
Thanks. so to modify flowviz to get epics & stories I need to change the jql queries to get hierarchyLevel in (0,1) for workitems completed, inprogress, history/2/5 & WIP 1/2 correct?

**Reply by Karloyski** [reply]
I'm sorry, what does "change jql queries" mean? remove this from the source code?

**Reply by nbrown02** [reply]
In the "Source" in Power Query you will see the original queries used :)

**Comment by nbrown02** [comment]
Yup that's correct - I'd also double check the tables don't have any
'Filtered Rows' steps that look at hierarchy level :)

**Comment by MaximumTrainer** [comment]
Thanks. Interestingly when I change this the I'm still getting some hierarchy level 0 data in the datasets so will figure out why.
Related to this when the data refresh is triggered does this also update the Jira status table (which looks like its stored serialised data)

**Comment by nbrown02** [comment]
It should update all tables yes :) Sent from my iPhoneOn 5 Jun 2024, at 18:39, MaximumTrainer ***@***.***> wrote:﻿
Thanks. Interestingly when I change this the I'm still getting some hierarchy level 0 data in the datasets so will figure out why.
Related to this when the data refresh is triggered does this also update the Jira status table (which looks like its stored serialised data)
